src/JuMP.jl

```diff
@@ -308,16 +285,34 @@ backend(model::Model) = model.moi_backend
 Return mode (Direct, Automatic, Manual) of model.
 """
 function mode(model::Model)
-    if !(backend(model) isa MOI.Bridges.LazyBridgeOptimizer{<:MOIU.CachingOptimizer} ||
-         backend(model) isa MOIU.CachingOptimizer)
+    if !(backend(model) isa MOIU.CachingOptimizer)
         return Direct
-    elseif caching_optimizer(model).mode == MOIU.Automatic
+    elseif backend(model).mode == MOIU.Automatic
         return Automatic
     else
         return Manual
     end
 end
 
+"""
+    bridge_constraints(model::Model)
+
+Return a `Bool` indicating whether the model `model` is in manual or automatic
```

Comment:
The wording is a bit confusing. Maybe split into cases. When in manual or automatic mode, returns X. When in direct mode, returns false.